### PR TITLE
hack to allow physical file provider to be serialized using json

### PR DESCRIPTION
**Pull request by jason-bragg**

### CODE REVIEW

File: test/TestInfrastructure/TestExtensions/TestDefaultConfiguration.cs [hunk 1]
```diff
@@ -48,6 +52,39 @@ public static void ConfigureHostConfiguration(IConfigurationBuilder builder)
...
         }
 
+        /// <summary>
+        /// Hack, allowing PhysicalFileProvider to be serialized using json
+        /// </summary>
+        [Serializable]
+        [JsonObject(MemberSerialization.OptIn)]
+        private class MyPhysicalFileProvider : IFileProvider
+        {
+            private PhysicalFileProvider fileProvider;
```
Comment (ReubenBond) [inline test/TestInfrastructure/TestExtensions/TestDefaultConfiguration.cs]: Alternatively/additionally, this could have `[NonSerialized]` on it.

**Comment by xiazen** [comment]
just curious, why is this change?

**Comment by jason-bragg** [comment]
It seems the upgrade of Microsoft.Configuration included changes to PhysicalFileProvider which made it no longer serializable via json.  It's not clear that it was ever supposed to be serializable.  It's likely we're using the configuration sources incorrectly.  That's why this is a bit of a hack.  It just works around the issue to unblock our testing, but we may want to reconsider serializing configuration sources.

**Comment by jason-bragg** [comment]
oops.. hit close instead of comment.. bleh...

File: test/TestInfrastructure/TestExtensions/TestDefaultConfiguration.cs [hunk 2]
```diff
@@ -48,6 +52,39 @@ public static void ConfigureHostConfiguration(IConfigurationBuilder builder)
             builder.AddEnvironmentVariables("Orleans");
         }
 
+        /// <summary>
+        /// Hack, allowing PhysicalFileProvider to be serialized using json
+        /// </summary>
+        [Serializable]
+        [JsonObject(MemberSerialization.OptIn)]
+        private class MyPhysicalFileProvider : IFileProvider
```
Comment (jdom) [inline test/TestInfrastructure/TestExtensions/TestDefaultConfiguration.cs]: nit: I'd call it something more explicit, such as `SerializablePhysicalFileProvider`

**Comment by xiazen** [comment]
@jason-bragg opps seems my change did this....

**Pull request merged**